ddtrace/tracer/textmap_test.go

```diff
@@ -2429,3 +2430,64 @@ func TestMalformedTID(t *testing.T) {
 		assert.Equal(t, "640cfd8d00000000", root.Meta[keyTraceID128])
 	})
 }
+
+func BenchmarkComposeTracestate(b *testing.B) {
+	ctx := new(spanContext)
+	ctx.trace = newTrace()
+	ctx.origin = "synthetics"
+	ctx.trace.setPropagatingTag("_dd.p.keyOne", "json")
+	ctx.trace.setPropagatingTag("_dd.p.KeyTwo", "123123")
+	ctx.trace.setPropagatingTag("_dd.p.table", "chair")
+	ctx.isRemote = false
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		composeTracestate(ctx, 1, "s:-2;o:synthetics___web")
+	}
+}
+
+func TestStringMutator(t *testing.T) {
+	sm := &stringMutator{}
+	rx := regexp.MustCompile(",|~|;|[^\\x21-\\x7E]+")
+	tc := []struct {
+		name  string
+		input string
+	}{
+		{
+			name:  "empty",
+			input: "",
+		},
+		{
+			name:  "no special characters",
+			input: "abcdef",
+		},
+		{
+			name:  "special characters",
+			input: "a,b;c~~~~d;",
+		},
+		{
+			name:  "special characters and non-ascii",
+			input: "a,b👍👍👍;c~d👍;",
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			expected := rx.ReplaceAllString(tt.input, "_")
+			actual := sm.Mutate(originDisallowedFn, tt.input)
+			assert.Equal(t, expected, actual)
+		})
+	}
+}
+
+func FuzzStringMutator(f *testing.F) {
+	rx := regexp.MustCompile(",|~|;|[^\\x21-\\x7E]+")
+	f.Add("a,b;c~~~~d;")
+	f.Add("a,b👍👍👍;c~d👍;")
+	f.Fuzz(func(t *testing.T, input string) {
+		sm := &stringMutator{}
+		expected := rx.ReplaceAllString(input, "_")
+		actual := sm.Mutate(originDisallowedFn, input)
+		if expected != actual {
+			t.Fatalf("expected: %s, actual: %s", expected, actual)
```

Comment:
Nice use of fuzzing :) This fails for me with the following input:

```
go test fuzz v1
string("=")
```

and the test output is:

```
fuzz: elapsed: 0s, gathering baseline coverage: 0/2 completed
fuzz: elapsed: 0s, gathering baseline coverage: 2/2 completed, now fuzzing with 10 workers
fuzz: minimizing 752-byte failing input file
fuzz: elapsed: 1s, minimizing
--- FAIL: FuzzStringMutator (0.52s)
    --- FAIL: FuzzStringMutator (0.00s)
        textmap_test.go:2490: expected: =, actual: ~
    
    Failing input written to testdata/fuzz/FuzzStringMutator/b074d9d373d06c31
    To re-run:
    go test -run=FuzzStringMutator/b074d9d373d06c31
FAIL
exit status 1
FAIL	gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer	1.100s
```

It looks like the original regex was `,|~|;|[^\\x21-\\x7E]+`, i.e. no `=` matching. From the comment on `originDisallowedFn`, looks like replacing `=` with `~` is expected. So, perhaps add the `strings.ReplaceAll(oWithSub, "=", "~"))` from the original implementation when generating the expected output?